deebot_client/hardware/deebot/2pv572.py

```diff
@@ -0,0 +1,92 @@
+"""2pv572 Capabilities."""
+
+from __future__ import annotations
+
+from deebot_client.capabilities import (
+    Capabilities,
+    CapabilityClean,
+    CapabilityCleanAction,
+    CapabilityCustomCommand,
+    CapabilityEvent,
+    CapabilityExecute,
+    CapabilityLifeSpan,
+    CapabilitySetTypes,
+    CapabilityStats,
+    DeviceType,
+)
+from deebot_client.commands.json.custom import CustomCommand
+from deebot_client.commands.xml import (
+    Charge,
+    Clean,
+    CleanArea,
+    GetBatteryInfo,
+    GetCleanLogs,
+    GetCleanSpeed,
+    GetCleanState,
+    PlaySound,
+    SetCleanSpeed,
+)
+from deebot_client.commands.xml.charge_state import GetChargeState
+from deebot_client.commands.xml.error import GetError
+from deebot_client.commands.xml.stats import GetCleanSum
+from deebot_client.const import DataType
+from deebot_client.events import (
+    AvailabilityEvent,
+    BatteryEvent,
+    CleanLogEvent,
+    CustomCommandEvent,
+    ErrorEvent,
+    FanSpeedEvent,
+    FanSpeedLevel,
+    LifeSpanEvent,
+    NetworkInfoEvent,
+    ReportStatsEvent,
+    StateEvent,
+    StatsEvent,
+    TotalStatsEvent,
+)
+from deebot_client.models import StaticDeviceInfo
+from deebot_client.util import short_name
+
+from . import DEVICES
+
+DEVICES[short_name(__name__)] = StaticDeviceInfo(
+    DataType.XML,
+    Capabilities(
+        availability=CapabilityEvent(AvailabilityEvent, []),
+        battery=CapabilityEvent(BatteryEvent, [GetBatteryInfo()]),
+        charge=CapabilityExecute(Charge),
+        clean=CapabilityClean(
+            action=CapabilityCleanAction(command=Clean, area=CleanArea),
+            log=CapabilityEvent(CleanLogEvent, [GetCleanLogs()]),
+        ),
+        custom=CapabilityCustomCommand(
+            event=CustomCommandEvent, get=[], set=CustomCommand
+        ),
+        device_type=DeviceType.VACUUM,
+        error=CapabilityEvent(ErrorEvent, [GetError()]),
+        fan_speed=CapabilitySetTypes(
+            event=FanSpeedEvent,
+            get=[GetCleanSpeed()],
+            set=SetCleanSpeed,
+            types=(
+                FanSpeedLevel.NORMAL,
+                FanSpeedLevel.MAX,
+            ),
+        ),
+        life_span=CapabilityLifeSpan(
+            types=(),
+            event=LifeSpanEvent,
+            get=[],
+            reset=CustomCommand,
+        ),
```

Comment:
```suggestion
        life_span=CapabilityLifeSpan(
            types=(
                LifeSpan.BRUSH,
                LifeSpan.SIDE_BRUSH,
                LifeSpan.DUST_CASE_HEAP,
            ),
            event=LifeSpanEvent,
            get=[
                GetLifeSpan(LifeSpan.BRUSH),
                GetLifeSpan(LifeSpan.SIDE_BRUSH),
                GetLifeSpan(LifeSpan.DUST_CASE_HEAP),
            ],
            reset=CustomCommand,
        ),
```

This should already work, if your robot has this feature (tested with my deebot 900). But you need to adjust the imports.

Reply:
Thanks, let's see